openedx/core/djangoapps/agreements/models.py

```diff
@@ -0,0 +1,22 @@
+"""
+Agreements models
+"""
+
+from django.contrib.auth.models import User  # lint-amnesty, pylint: disable=imported-auth-user
+from django.db import models
+from model_utils.models import TimeStampedModel
+from opaque_keys.edx.django.models import CourseKeyField
+
+
+class IntegritySignature(TimeStampedModel):
+    """
+    This model represents an integrity signature for a user + course combination.
+
+    .. no_pii:
+    """
+    user = models.ForeignKey(User, db_index=True, on_delete=models.CASCADE)
```

Comment:
No need to do `on_delete` statement. edX do not delete user accounts by removing database row. We obfuscate it.

Reply:
I believe `on_delete` is required for foreign keys